PeerioTechnologies/peerio-desktop
# feat: zero chats new design

[PR guideline](https://github.com/PeerioTechnologies/peerio-icebear/blob/dev/docs/CONTRIBUTING.md)

### PR author

#### Relevant issue tracker links  
https://app.clubhouse.io/peerio/story/327/desktop-implement-zero-chat-state
https://peerio.tpondemand.com/entity/7735

#### Linked PRs in other repos
https://github.com/PeerioTechnologies/peerio-icebear/pull/87

#### Manual testing instructions  
*previously:*
New zero chat state! Should appear when user is in chats view and they have no DMs or Rooms. Should not appear if user has DMs or Rooms.

I had to alter the `padding-right` of the `.messages` class, which affects the chats view (rooms and DMs) and the channel invites view.

I also had to add a `z-index` property on the entire `AppNav` component (far left sidebar with the main app navigation) which theoretically could affect the entire app. I've verified that everything that's supposed to draw over the app still does.

*reopened, updates:*
The app's starting view has been completely rerouted. We now have a new Loading view, with a `reaction` to `chatStore.loaded`; on `loaded`, route either to Chats or ZeroChats depending on whether there are chats.

I've verified the following scenarios (`A` is existing user, `B` is new user with no chats yet).
* `B` is on ZeroChats view. `A` starts a chat with `B` --> `B`'s app automatically goes to Chats view
* `B` is on ZeroChats view. `A` invites `B` to a Room --> "View Room Invites" button appears in `B`'s ZeroChats view
  * ~~There's a bug that I believe exists already but couldn't find in TP. If `A` invites `B` to a room, then `A` removes `B` from that room (or deletes the room) before `B` can accept/decline the invitation, we get some errors in console but no UI feedback. On `B`'s side, `chatInviteStore.received.length` still shows as 1, so the "View Room invites" button remains and the invite itself remains, but nothing happens on clicking accept/deny.~~ Eren says this isn't happening anymore, yay

#### Additional notes
~~If user does have chats, there's a quick "flash" of this view on login, before the chatStore starts loading. Not sure what the best way is to tackle this.~~ fixed with new routing

Re: `z-index`-ing, we should probably standardize our z's. Not a lot of components need an explicitly set z at the moment so it's all kinda arbitrary, it seems.

----
### Reviewer

* Localisation.
* Architectural decisions.
* Performance.
* Comments.
* Big feature tech design documentation.
* Readability.
* DRY.
* Can we extract something to SDK?
* Are there any bicycles invented that should be replaced with existing SDK
* Are there unit and e2e tests added?

----
### Repository owner
